Expose audit rules configuration in the `shoot-rsyslog-relp` api (#149)
gardener/gardener-extension-shoot-rsyslog-relp: docs/usage/configuration.md

      apiVersion: rsyslog-relp.extensions.gardener.cloud/v1alpha1
      kind: RsyslogRelpConfig
      # Set the target server to which logs are sent. The server must support the RELP protocol.

>       target: some.rsyslog-relp.server

Reviewer: You could also fix similar typo in https://github.com/gardener/gardener-extension-shoot-rsyslog-relp/blob/c8be0cbeba2f36ead53222546f0a9b2e7edde1d8/pkg/cmd/rsyslogrelp/options.go#L5.

You can set a few additional parameters for the TLS connection: `.tls.authMode`, `tls.permittedPeer`, and `tls.tlsLib`. Refer to the rsyslog documentation for more information on these parameters:
- [`.tls.authMode`](https://www.rsyslog.com/doc/v8-stable/configuration/modules/omrelp.html#tls-authmode)
- [`.tls.permittedPeer`](https://www.rsyslog.com/doc/v8-stable/configuration/modules/omrelp.html#tls-permittedpeer)
- [`.tls.tlsLib`](https://www.rsyslog.com/doc/v8-stable/configuration/modules/imrelp.html#tls-tlslib)

> ### Configuring the Audit Daemon on the Shoot Nodes

Reviewer: Does it make sense to also add the new fields in the Shoot spec in https://github.com/gardener/gardener-extension-shoot-rsyslog-relp/blob/b6e6bb8a1091c3443d36efbfda3fb60bd18b23b8/docs/usage/configuration.md#shoot-configuration?

Alternatively, you can define your own `auditd` rules by using the following configuration:
```yaml
apiVersion: rsyslog-relp.extensions.gardener.cloud/v1alpha1
kind: Auditd
auditRules: |
  ## First rule - delete all existing rules
  -D
  ## Now define some custom rules
  -a exit,always -F arch=b64 -S setuid -S setreuid -S setgid -S setregid -F auid>0 -F auid!=-1 -F key=privilege_escalation
  -a exit,always -F arch=b64 -S execve -S execveat -F euid=0 -F auid>0 -F auid!=-1 -F key=privilege_escalation
```

> To deploy this configuration, it must be embedded in a config map. **Note** that the data key storing this configuration must be named `auditd`.
> An example `ConfigMap` is given below:

Reviewer: ```suggestion
To deploy this configuration, it must be embedded in an immutable ConfigMap.

> [!NOTE]
> The data key storing this configuration must be named `auditd`.

An example `ConfigMap` is given below:
```

    auditRules: |
      ## First rule - delete all existing rules
      -D
      ## Now define some custom rules
      -a exit,always -F arch=b64 -S setuid -S setreuid -S setgid -S setregid -F auid>0 -F auid!=-1 -F key=privilege_escalation
      -a exit,always -F arch=b64 -S execve -S execveat -F euid=0 -F auid>0 -F auid!=-1 -F key=privilege_escalation
```

After creating such a `ConfigMap`, it must be included in the Shoot's `spec.resources` array and then referenced from the `providerConfig.auditConfig.configMapReferenceName` field in the `shoot-rsyslog-relp` extension configuration.

An example configuration is given below:

```yaml
kind: Shoot
metadata:
  name: bar
  namespace: garden-foo
...
spec:
  extensions:
  - type: shoot-rsyslog-relp
    providerConfig:
      apiVersion: rsyslog-relp.extensions.gardener.cloud/v1alpha1
      kind: RsyslogRelpConfig
      target: some.rsyslog-relp.server
      port: 10250
      loggingRules:
      - severity: 7
      auditConfig:
        enabled: true
        configMapReferenceName: audit-config
  resources:
    - name: audit-config
      resourceRef:
        apiVersion: v1
        kind: ConfigMap
        name: audit-config-v1
```

> Finally, by setting `providerConfig.auditConfig.enabled` to `false` in the `shoot-rsyslog-relp` extension configuration, you can completely disable the additional audit configurations performed by the extension.

Reviewer: IIUC, there are audit rules coming from 3 sources: rules that come with the host OS, rules added by default by the extension and rules that can now be added by the Shoot owners.

When `providerConfig.auditConfig.enabled` won't deploy any rules (extension default ones and Shoot owners ones), right? We can be more explicit here as 
> [...] you can completely disable the additional audit configurations performed by the extension.

it is not super clear what you disable.

Reviewer: Just to reply to myself:
- The following deploys the extension default audit rules as of today
```yaml
      auditConfig:
        enabled: true
```

- The following deploys only the rules specified in the ConfigMap
```yaml
      auditConfig:
        enabled: true
        configMapReferenceName: audit-config
```

- The following does not deploy any audit rules
```yaml
      auditConfig:
        enabled: false
        configMapReferenceName: audit-config
```

Author: Good point, I could actually list the options like that. I tried simply modifying the text but this is clearer, thanks!
